src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -81,8 +83,9 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
 
         private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, DiagnosticBag diagnostics)
         {
-            Debug.Assert(node != null);
-            Binder switchBinder = this.GetBinder(node);
+            RoslynDebug.Assert(node is { });
```

Comment:
>is { } [](start = 36, length = 6)

Jared has been driving some conventions around null checks as part of nullability adoption. We've aligned on `is object` as preferred for non-null check. This applies here and numerous other places which I won't individually tag.

Reply:
I prefer not to use a type check as a null check or set what I consider to be a bad example for the public.  When C# 9 is final, I'll go through and change these to `is not null`.

---
In reply to: [380884849](https://github.com/dotnet/roslyn/pull/41503#discussion_r380884849) [](ancestors = 380884849)

Reply:
Please discuss this with Jared so that we're on the same page in terms of compiler coding guidelines.

---
In reply to: [381588860](https://github.com/dotnet/roslyn/pull/41503#discussion_r381588860) [](ancestors = 381588860,380884849)

Reply:
Jared has already stated that he doesn't want to have compiler coding guidelines separate than the common dotnet guidelines, and in any case he doesn't want to impose guidelines himself unless there is a deep rift that the group is unable to resolve itself.  Is that the situation we're in?

---
In reply to: [384026995](https://github.com/dotnet/roslyn/pull/41503#discussion_r384026995) [](ancestors = 384026995,381588860,380884849)

Reply:
Briefly mentioned it to him this morning. He didn't seem to care about uniformity as much as I expected.
Agreed this problem goes away once we have `is not null` :-)
Thanks